PyInstaller/hooks/hook-PySide2.QtWebEngineWidgets.py

```diff
@@ -1,51 +1,101 @@
-            (os.path.join(resources_dir, 'qtwebengine_resources.pak'), '.'),
-            # The distributed Info.plist has LSUIElement set to true, which prevents the
-            # icon from appearing in the dock.
-            (os.path.join(libdir, 'QtWebEngineCore.framework', 'Versions', '5',
-                          'Helpers', 'QtWebEngineProcess.app', 'Contents', 'Info.plist'),
-             os.path.join('QtWebEngineProcess.app', 'Contents'))
+            # Gather translations needed by Chromium.
+            (os.path.join(pyside2_locations['TranslationsPath'],
+                          locales),
+             os.path.join(*(rel_data_path + ['translations',
+                                             locales]))),
+            # Per the `docs
+            # <https://doc.qt.io/qt-5.10/qtwebengine-deploying.html#deploying-resources>`_,
+            # ``DataPath`` is the base directory for ``resources``.
+            #
+            # When Python 3.4 goes EOL (see `PEP 448`_, this is better written
+            # as ``os.path.join(*rel_data_path, resources)``.
+            (os.path.join(pyside2_locations['DataPath'], resources),
+             os.path.join(*(rel_data_path + [resources]))),
+            # Include the webengine process. The ``LibraryExecutablesPath``
+            # is only valid on Windows and Linux.
+            #
+            # Again, rewrite when Python 3.4 is EOL to
+            # ``os.path.join(*rel_data_path, remove_prefix(...``.
+            (os.path.join(pyside2_locations['LibraryExecutablesPath'],
+                          'QtWebEngineProcess*'),
+             os.path.join(*(rel_data_path + [get_relative_path_if_possible(
+                            pyside2_locations['LibraryExecutablesPath'],
+                            pyside2_locations['PrefixPath'] + '/')])))
         ]
+
+    # Add Linux-specific libraries.
+    if compat.is_linux:
+        # The automatic library detection fails for `NSS
+        # <https://packages.ubuntu.com/search?keywords=libnss3>`_, which is
+        # used by QtWebEngine. In some distributions, the ``libnss``
+        # supporting libraries are stored in a subdirectory ``nss``. Since
+        # ``libnss`` is not statically linked to these, but dynamically loads
+        # them, we need to search for and add them.
+        #
+        # First, get all libraries linked to ``PyQt5.QtWebEngineWidgets``.
+        for imp in getImports(
+                get_module_file_attribute('PySide2.QtWebEngineWidgets')):
+            # Look for ``libnss3.so``.
+            if os.path.basename(imp).startswith('libnss3.so'):
+                # Find the location of NSS: given a ``/path/to/libnss.so``,
+                # add ``/path/to/nss/*.so`` to get the missing NSS libraries.
+                nss_subdir = os.path.join(os.path.dirname(imp), 'nss')
+                if os.path.exists(nss_subdir):
+                    binaries.append((os.path.join(nss_subdir, '*.so'), 'nss'))
```

Comment:
Isn't libnss part of glibc and thus should be installed on every Linux system anyway?

Reply:
I added this code in the original hook in af23dcceb3f997bda40b1dd5c79cbe61feea63b2. I think this fixed failures on Travis.